xtext-website/documentation/103_domainmodelnextsteps.md

```diff
@@ -189,6 +189,47 @@ class DomainmodelGenerator extends AbstractGenerator {
 
 If you want to play around with Xtend, you can try to use the Xtend tutorial which can be materialized into your workspace. Simply choose *New &rarr; Example &rarr; Xtend Examples &rarr; Xtend Introductory Examples* and have a look at the features of Xtend. As a small exercise, you could implement support for the *many* attribute of a *Feature* or enforce naming conventions, e.g. generated field names should start with an underscore.
 
+## Creating Custom Validation Rules {#tutorial-validation}
+
+One of the main advantages of DSLs is the possibility to statically validate domain-specific constraints. Because this is a common use case, Xtext provides a dedicated hook for this kind of validation rules. In this lesson, we want to ensure that the name of an *Entity* starts with an upper-case letter and that all features have distinct names across the inheritance relationship of an *Entity*.
+
+Locate the class *DomainmodelValidator* in the package *org.example.domainmodel.validation* of the language project. Defining the constraint itself is only a matter of a few lines of code:
+
+```java
+public static final String INVALID_NAME = "invalidName";
+
+@Check
+public void checkNameStartsWithCapital(Entity entity) {
+    if (!Character.isUpperCase(entity.getName().charAt(0))) {
+        warning("Name should start with a capital", 
+            DomainmodelPackage.Literals.TYPE__NAME,
+            INVALID_NAME);
+    }
+}
+```
+
+Any name for the method will do. The important thing is the [Check]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/validation/Check.java) annotation that advises the framework to use the method as a validation rule. If the name starts with a lower case letter, a warning will be attached to the name of the *Entity*.
+
+The second validation rule is straight-forward, too. We traverse the inheritance hierarchy of the *Entity* and look for features with equal names.
+
+```java
+@Check
```

Comment:
You might like to add a note about the `CheckType` if it is not already covered in the document.

Reply:
I have changed this now

Reply:
Couldn't find it in the tutorials so now I've included a note about it too